rbac/server/api/roles.py

```diff
@@ -63,12 +64,15 @@
     if not response:
         txn_key, txn_user_id = await utils.get_transactor_key(request)
         role_id = str(uuid4())
+        metadata1 = request.json.get("metadata")
+        metadata1 = {"sync_direction": "OUTBOUND"}
+        metadata1_str = json1.dumps(metadata1)
```

Comment:
How come we are now saving the metadata field as a string instead of a dictionary? I believe you were looking into the protobufs for saving dictionaries? I also found this resource that states protobufs also support maps data structure: https://developers.google.com/protocol-buffers/docs/proto3#maps

Reply:
Consideration 2: The metadata field & metadata table schema should be untouched, as per the product owner.
So, storing the new sync_direction flag is stored in the metadata field as string, as earlier.